open_spiel/python/pybind11/pybind11.h

```diff
@@ -31,7 +31,6 @@
 #include "pybind11/include/pybind11/cast.h"
 #include "pybind11/include/pybind11/detail/common.h"
 #include "pybind11/include/pybind11/detail/descr.h"
-#include "pybind11/include/pybind11/detail/smart_holder_type_casters.h"
```

Comment:
Removing this private header makes you forward compatible (smart_holder branch before and after https://github.com/pybind/pybind11/pull/5257 was merged).

If you don't care about backward compatibility (now or later) and also want to clean up:

Also remove all
* pybind11/include/pybind11/smart_holder.h includes
* macros you see here: https://github.com/pybind/pybind11/blob/48f25275c44d52d0ceade122e328dc1f2e48ef44/include/pybind11/smart_holder.h

As of 3cdecf7b2c88095eec13a49c13b768cf7a82189b, you're only using the now obsolete `PYBIND11_SMART_HOLDER_TYPE_CASTERS` macro (37 in 15 files).

Reply:
Ok great, I'll do the cleanup in a follow-up PR (or internally) over the next week or so.